opti/tools/reduce.py

```diff
@@ -0,0 +1,271 @@
+from logging.handlers import WatchedFileHandler
+import numpy as np
+import pandas as pd                                                     
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+
+from opti import Problem
+from opti.constraint import Constraints, LinearEquality
+from opti.objective import Objectives
+from opti.parameter import Categorical, Parameters
+from sympy import Matrix
+
+
+#TODO: Wie geht man mit diskreten inputs um? --> einfach domain beibehalten?
+#TODO: funktion, die das Modell f für das reduzierte problem anpasst
+#TODO: Was bedeutet das attribut Models in class Problem?
```

Comment:
It's functionality to add known relationships between inputs and outputs, such as cost = c1 * x1 + c2 * x2 ...